### PR TITLE
Added modal for creating new templates

### PR DESCRIPTION
**Pull request by bryanansong**
## Description
I created a modal component for giving the user some options to create a workout template. This information is passed as a POST request to the backend. I made routes to pass this data to my workoutTemplatesController file where I have a method to create a WorkoutTemplate record along with an Exercise and a WorkoutTemplateExercise record. 
<what this pull request is doing>
<what will be done in later PRs and not included here>

## Test Plan

https://github.com/bryanansong/ZenFitness/assets/50881304/e3c92658-896b-41b8-8dc4-e87458fd8670

<insert images or gifs of feature>
<otherwise, say how code was tested if not UI facing>
<any edge cases you might have specifically tested>

### CODE REVIEW

File: frontend/src/components/WorkoutTemplateModal/WorkoutTemplateModal.tsx [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    }
+    setSearchTerm("");
+  };
+
+  const handleRemoveExercise = (exercise: string) => {
+    setSelectedExercises((prev) => prev.filter((ex) => ex !== exercise));
+  };
+
+  const handleCreateTemplate = async () => {
+    if (!workoutName || selectedExercises.length === 0) {
```
Comment (Wang96) [inline frontend/src/components/WorkoutTemplateModal/WorkoutTemplateModal.tsx:58]: The default value for workoutName is empty string, not quite familiar with typescript but will !workoutName also check for empty string or just null?

**Comment by Wang96** [comment]
A general question for template creation. I think in your template structure you will include isPublic, duration or count, but I didn't see these fields in the creation flow yet. Will this be added later?

**Comment by m5matthew** [comment]
Code looks great! For the demo—I have the same questions as Nianyi above (about isPublic, duration, count). Also, for template creation, do we let them specify how many sets/reps they should do?

File: frontend/src/components/WorkoutTemplateModal/WorkoutTemplateModal.tsx (hunk 1 above)
Comment (bryanansong) [reply frontend/src/components/WorkoutTemplateModal/WorkoutTemplateModal.tsx:58]: !workoutName checks for both empty string and null values

**Comment by bryanansong** [comment]
For template creation, I recently added a field for specifying isPublic in a new PR. Duration, sets and reps are specified when creating a workout session which I am yet to work on. For template creation, users will not specify how many set/reps they'll do. They can decide that when they are logging their workout session
